FlashCap.Core/Devices/DirectShowDevice.cs

```diff
@@ -320,4 +338,15 @@ protected override Task OnStopAsync(CancellationToken ct)
         long timestampMicroseconds, long frameIndex,
         PixelBuffer buffer) =>
         buffer.CopyIn(this.pBih, pData, size, timestampMicroseconds, frameIndex, this.transcodeIfYUV);
+
+    protected override void SetControlProperty(CameraControlProperty property, int value)
+    {
+        lock (this)
+        {
+            if (this.IsRunning)
+            {
+                cameraControl.Set(property, value, CameraControlFlags.None);
```

Comment:
This call to `Set()` corresponds to a cross-apartment call on COM interface, so a procedure over the apartment is required. Use `this.workingContext.InvokeAsync()` to ensure that it is called on the correct thread. The implementation of the `OnStartAsync()` method will be helpful.
Perhaps it needs to be asynchronized, like `SetControlPropertyAsync()`.